### PR TITLE
MAINT: remove optimize.anneal for 0.16.0 (was deprecated in 0.14.0).

### LINKED ISSUE #1653: scipy.optimize.anneal does not respect lower/upper bounds (Trac #1126)
_Original ticket http://projects.scipy.org/scipy/ticket/1126 on 2010-03-04 by trac user lboussouf, assigned to unknown._

Running the code below raises an exception as anneal try to evaluate f at 1.97065187015 which is forbidden as specified in bounds.

If we look at line 106 in the code : [http://projects.scipy.org/scipy/browser/trunk/scipy/optimize/anneal.py]

It's obvious why it does not respect bounds : xc is designed to respect them, but as we do xnew = x0 + xc the following line, there is no chance to respect bounds.

And the same problem appears in all update_guess subroutines.

Is anybody supporting that code or would it be better to handle it myself ? I can't stand thinking people are using this code with such mistakes. My optimizer soul is hurt ...

    import scipy.optimize
    
    # Define function
    def f(x):
        print x
        if x < -1 or x > 1:    	
        	raise Exception
        else:
            return x**2
    
    
    # Solve minimization problem using SA
    x = scipy.optimize.anneal(func=f,x0=0,lower=-1,upper=1)

### PR DESCRIPTION
**Pull request by rgommers**
Closes gh-1653, gh-4042, gh-4009, gh-1402.

Also add `basinhopping` and `differential_evolution` links in a few places.

### CODE REVIEW

File: doc/source/tutorial/examples/5-1 [hunk 1]
```diff
@@ -32,8 +32,8 @@ Optimization Tools
 
   Global Optimizers
 
-   anneal       --  Simulated Annealing
    basinhopping --  Basin-hopping stochastic optimizer
+   differential_evolution -- Stochasting optimizer
```
Comment (argriffing) [inline doc/source/tutorial/examples/5-1]: spelling

**Comment by dlax** [comment]
Fixed the typo above and merged (by hand). Thanks!

**Comment by rgommers** [comment]
merge commit: 2c8f2e6

**Comment by rgommers** [comment]
thanks @dlax

**Pull request closed without merging**